omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,246 @@
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		} else if (rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+			description.addSelfLink();
+		}
+		return description;
+	}
+	
+	@PropertyGetter("location")
+	public String getLocation(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getLocation().getUuid();
+	}
+	
+	@PropertyGetter("identifierType")
+	public String getIdentifierType(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType().getUuid();
+	}
+	
+	@PropertyGetter("source")
+	public String getSource(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getSource().getUuid();
+	}
+	
+	@PropertyGetter("uuid")
+	public String getUuid(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getUuid();
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType() + " - " + autoGenerationOption.getSource() + " - "
+		        + autoGenerationOption.getLocation();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("identifierType");
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		Object getManualEntryEnabled = (postBody.get("manualEntryEnabled"));
+		Object getAutomaticallyEnabled = postBody.get("automaticGenerationEnabled");
+		Object getSource = postBody.get("source");
+		Object getLocation = postBody.get("location");
+		Object getIdentifierType = postBody.get("identifierType");
+		String manualEntryEnabled = ((getManualEntryEnabled != null) && (!getManualEntryEnabled.equals("")))
+		        ? getManualEntryEnabled.toString()
```

Comment:
Have you looked at how @adamsdenniskariuki deals with the check for null and empty string?